packages/gatsby-cli/src/reporter/reporter.ts

```diff
@@ -136,7 +195,8 @@ class Reporter {
       }
     }
 
-    const structuredError = constructError({ details })
+    const structuredError = constructError({ details }, this.getErrorMap())
```

Comment:
Can we do the errorMap dance here so we don't have to spread each time?

```suggestion
    let errorMapEntry = defaultError;
    if (details.id) {
      // Look at original errorMap, ids cannot be overwritten
      if (errorMap[details.id]) {
        errorMapEntry = errorMap[details.id]
      }
  
      if (this.errorMap[details.id]) {
        errorMapEntry = this.errorMap[details.id]
      }
    }
    
    const structuredError = constructError({ details }, errorMapEntry)
```